### PR TITLE
Restore variable encumbrance

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: Features "Restore variable encumbrance for non-rigid containers"

#### Purpose of change
One feature that got lost in the nested containers implementation was that non-rigid worn items should change their encumbrance with volume.

#### Describe the solution
Reinstate that feature.

It makes more sense than it did previously because now we know how full each individual item is, rather than having to approximate based on the overall inventory usage.

Each 250ml adds one unit of encumbrance, unless an explicit `max_encumbrance` value is given, in which case the encumbrance interpolates between the two endpoints.

#### Describe alternatives you've considered
Requiring `max_encumbrance` to be given.

#### Testing
Restored the `iteminfo` unit tests which cover this case (they are no longer `[!mayfail]`) and update the test data accordingly.

#### Additional context
I was getting annoyed by the expected failures in the tests.

### CODE REVIEW

**Comment by jbytheway** [comment]
I see @wapcaplet (#40314) and I have trodden on each others toes a bit here.  I need to rebase now.

**Comment by KorGgenT** [comment]
looks like json needs linting

File: src/item_contents.cpp [hunk 1]
```diff
@@ -786,6 +787,40 @@ int item_contents::remaining_capacity_for_liquid( const item &liquid ) const
...
+        nonrigid_capacity += pocket.volume_capacity();
+    }
+    assert( nonrigid_volume <= nonrigid_capacity );
+    if( nonrigid_capacity == 0_ml ) {
+        return 0;
+    }
+    return nonrigid_volume * 1.0 / nonrigid_capacity;
+}
+
+bool item_contents::is_rigid() const
```
Comment (KorGgenT) [inline src/item_contents.cpp:811]: i think maybe this should be renamed to any_pocket_is_rigid() just to make sure it's used properly by any additional contributors?

**Comment by dissociativity** [comment]
maybe cargo pants can finally be less encumbering than many armours with this eh?

**Comment by jbytheway** [comment]
I've realised that this is actually a fairly substantial behaviour change because previously the variable encumbrance only applied to things that either had an explicit `max_encumbrance` or directly contained objects (like the waterskins).  Now, because everything directly contains objects, all clothing that *doesn't* specify `max_encumbrance` will start accumulating additional encumbrance, where it didn't before.  So, e.g. cargo pants as mentioned by @dissociativity will be *more* encumbering after this change (used to be 16; now between 16 and 32).

We probably need a review of the encumbrance of all those items.  Or maybe we can think of a better default behaviour than what I've currently implemented for items lacking a `max_encumbrance` field.  Thoughts?

**Comment by jbytheway** [comment]
There are some issues in the tests I need to get to the bottom of, so marking as draft.  I'd still appreciate thoughts on how to deal with increased encumbrance for existing items, though.  In particular, is it acceptable to merge this first and fix the items later.

**Comment by dissociativity** [comment]
I'm sure we can fix it in items, I'll happily go through some after this is merged!

**Comment by Termineitor244** [comment]
I would say that there is no problem with fixing the items later, many probably will need some thinking to balance their encumbrance numbers, cargo pants being a great example, in the future it would be better to drop the base encumbrance of all the items with several pockets so as to not end with an insane encumbrance...

That said, in the meantime the cargo pants will be basically useless haha, I think those could do better with a max_encumbrance and a reduced base encumbrance.

In another note, many players will be annoyed by the sudden change in encumbrance of their regular clothes, so a little warning could do.

**Comment by jbytheway** [comment]
I've written a script which does a decent job of identifying the items that need updating.  Hopefully I'll be able to adapt it to perform a minimally reasonable update to these items so that encumbrance doesn't go too crazy as soon as this is merged.

**Comment by Termineitor244** [comment]
Great! Thanks!

**Comment by jbytheway** [comment]
I got the automated script working.  For all the things that appeared to be "regular storage" clothing and didn't already have a `max_encumbrance` value I've set their new `max_encumbrance` to be their old `encumbrance` and their new `encumbrance` to be `max( max_encumbrance / 2, max_encumbrance - capacity / 250ml )`.

I think that should suffice for a first pass, but people will need to keep an eye out for items with silly values.

Then there was a bug with quiver volumes that took me several days to track down.  Now finally fixed, I believe.

Along the way I enhanced the iteminfo tests (changes I made to debug that quiver issue, but which are probably a good idea regardless).

**Pull request merged**